### PR TITLE
Update CI to test more version

### LINKED ISSUE #219: Dependency check
@hayesgb I'm cleaning up some things and noticed some dependencies that might be unused. Can you confirm that all of these are unnecessary?

**From setup.py**

* aiohttp
* msrestazure
* requests

**From dev-requirements.txt**

* aiohttp
* msrestazure
* scipy
* versioneer (my understanding is that installing versioneer is a one-time thing. Other contributors don't need it installed separately).

**Other**

* The file requirements.txt

### PR DESCRIPTION
**Pull request by TomAugspurger**
Test against the earliest, latest, and development versions of fsspec.

Closes #219. xref https://github.com/dask/s3fs/pull/462#issuecomment-821123657.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Hmm, CI isn't running. I must have messed up the pipelines file.

File: setup.py [hunk 1]
```diff
@@ -29,13 +29,11 @@
     long_description=open("README.md").read() if exists("README.md") else "",
     install_requires=[
         "aiohttp",
-        "azure-core>=1.5.0",
+        "azure-core>=1.7.0",
         "azure-datalake-store>=0.0.46,<0.1",
         "azure-identity",
-        "azure-storage-blob",
+        "azure-storage-blob>=12.5.0",
```
Comment (TomAugspurger) [inline setup.py:35]: This is required by the use of BlobClient.exists, and the import from `_list_blobs_helper`.

**Comment by TomAugspurger** [comment]
@hayesgb all green! Does this strategy (earliest, latest, dev) seem good to you?

I'd also like to add a nightly job to test against fsspec, to pick up if there are breaking changes between commits to adlfs. Maybe as a followup.

**Comment by hayesgb** [comment]
Hi @TomAugspurger this outlined strategy look great to me.  Appreciate all the help with this!

**Pull request merged**